src/HotChocolate/Core/src/Types/Configuration/ReadOnlySchemaOptions.cs

```diff
@@ -63,6 +66,12 @@ public ReadOnlySchemaOptions(IReadOnlySchemaOptions options)
         /// </summary>
         public bool UseXmlDocumentation { get; }
 
+        /// <summary>
+        /// Defines the name of the XML documenatation file to be read.
+        /// Only used if <seealso cref="UseXmlDocumentation"/> is true.
+        /// </summary>
+        public string XmlDocumentationFileName { get; }
```

Comment:
I understand that, but if we put something in the public API it must cater to more than just one use case. I think in most case people have a filename that is based on the assembly. Having a delegate there would allow your use-case as well as all the others.

essentially in your use case you would just pass in `assembly => "MyStaticFileName.xml"`.

Reply:
Also respect nullability ... so the property should. be nullable.

Reply:
I'm merely an outside contributor so if you feel that's the approach the library should take I'll update the PR to implement that, but personally I feel requiring them to specify a delegate which takes an Assembly reference is more specialized than a string property. If they need/want to use the assembly name, they can pass `Assembly.GetExecutingAssembly().Name` or whatever else in their configuration code. Or is the concern that the code doing the initialization may not be the assembly that gets provided at runtime?

Reply:
Types in schemas can be spread to multiple assemblies. Many projects I have been working on have their types not only in the executing assembly. With the implementation you provided the name you provide would apply to all documentation files from all assemblies out of which you build the GraphQL server. If using the delegate you can dynamically generate that documentation file. In many cases it would be something like `assembly => assembly.GetName().Name + ".xml"`

Reply:
Looking at your comment it seems that you expect the naming convention to look only at a single assembly. It actually traverses all assemblies that are used to conceive a schema.

Reply:
Ah, thank you for that clarification. I went back and looked through the calling code and I see what you mean now about needing to be able to pull data from multiple assemblies. It didn't occur to me that the compiler didn't combine xml docs into a single file.

I'll update the PR to use a delegate so callers have access to the assembly in question at runtime.

Reply:
awesome